src/Functions/errorCode.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Functions/castTypeToEither.h>
+#include <Functions/IFunctionImpl.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnString.h>
+#include <Core/Field.h>
+#include <unordered_map>
+
+#if !defined(ARCADIA_BUILD)
+#    include <Common/config_version.h>
+#endif
+
+extern std::unordered_map<int, const char *> auto_error_codes;
```

Comment:
Unordered map is using suboptimal hash function that will eat all the performance.

Reply:
The lookup in unordered_map is also suboptimal.